test/e2e/declarative_test_fwk/helper.py

```diff
@@ -201,16 +282,59 @@ def assert_items(self, expectations: dict, state: dict) -> None:
         if not expectations:
             # nothing to assert as there are no expectations
             return
-        if not state:
-            # there are expectations but no given state to validate
+        if not k8s_resource:
+            # there are expectations but no given k8s_resource state to validate
             # following assert will fail and assert introspection will provide useful information for debugging
-            assert expectations == state
+            assert expectations == k8s_resource
 
         for (key, value) in expectations.items():
             # conditions are processed separately
             if key == "conditions":
                 continue
-            assert (key, value) == (key, state.get(key))
+            # tags are processed by custom logic
+            if key == "tags":
+                self.assert_tags(value, k8s_resource.get(key))
+                continue
+            # assert any specific extra fields
+            if self.assert_extra_items_k8s(key, value, k8s_resource):
+                continue
+            assert value == k8s_resource.get(key)
+
+    def assert_extra_items_k8s(self, expected_name, expected_value, k8s_resource) -> bool:
+        """Asserts extra fields from k8s resource against fields from expectations
+        Override it as needed for custom verifications
+        Return true if the expected_name is asserted successfully
```

Comment:
I think there is room for improvement in this documentation on what the return value should be; it doesn't make sense to say that "expected_name is asserted successfully".  Can you describe more clearly what are the conditions for which this return value should be `True`?

Reply:
Explained it with clearer words.